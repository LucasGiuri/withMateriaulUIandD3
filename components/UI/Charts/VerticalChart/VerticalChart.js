import {BarChart, Bar, XAxis, YAxis, CartesianGrid, 
  Tooltip, ResponsiveContainer } from 'recharts';
import { SECONDARY } from '../../../../public/styles/colors';

const VerticalChartComponent = ({data}) => (
  <div style={{ width: '100%', height: 800 }}>
    <ResponsiveContainer>
      <BarChart
        layout="vertical"
        data={data}
      >
        <CartesianGrid stroke="#f5f5f5" />
        <XAxis type="number" />
        <YAxis dataKey="name" type="category" width={450} />
        <Tooltip />
        <Bar dataKey="value" barSize={20} fill={SECONDARY} />
      </BarChart>
    </ResponsiveContainer>
  </div>
);

export default VerticalChartComponent;
