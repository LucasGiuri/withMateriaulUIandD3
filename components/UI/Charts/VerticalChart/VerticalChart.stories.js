import { storiesOf } from '@storybook/react';
import BarChart from './BarChart';

const data = [
  {name: 'Shalion 1', value: 1},
  {name: 'Shalion 2', value: 2},
  {name: 'Shalion 3', value: 3},
  {name: 'Shalion 4', value: 4},
  {name: 'Shalion 5', value: 4},
];

storiesOf('BarChart')
  .add('BarChart default', () => (
    <BarChart data={data}/>
  ))
