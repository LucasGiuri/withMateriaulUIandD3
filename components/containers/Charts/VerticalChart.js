import PropTypes from 'prop-types';
import { Fragment } from "react";
import VerticalChartComponent from "../../UI/Charts/VerticalChart/VerticalChart";
import Title from "../../UI/Title/Title";

const VerticalChart = ({data}) => {
  let newData = [
    {id: "Amazon", data: []},
    {id: "Carethy", data: []},
    {id: "Carrefour", data: []},
    {id: "Sephora", data: []},
    {id: "El corte inglÃ©s", data: []},
  ];

  data.map(product => {
    const { name, avgPriceAmazon, count } = product;
    newData[0].data.push({name: name, value: avgPriceAmazon !== 0 ? avgPriceAmazon / count: 0})
  });

  data.map(product => {
    const { name, avgPriceCarethy, count } = product;
    newData[1].data.push({name: name, value: avgPriceCarethy !== 0 ? avgPriceCarethy / count: 0})
  });

  data.map(product => {
    const { name, avgPriceCarrefour, count } = product;
    newData[2].data.push({name: name, value: avgPriceCarrefour !== 0 ? avgPriceCarrefour / count: 0})
  });

  data.map(product => {
    const { name, avgPriceSephora, count } = product;
    newData[3].data.push({name: name, value: avgPriceSephora !== 0 ? avgPriceSephora / count: 0})
  });

  data.map(product => {
    const { name, avgPriceElcorteingles, count } = product;
    newData[4].data.push({name: name, value: avgPriceElcorteingles !== 0 ? avgPriceElcorteingles / count: 0})
  });

  return (
    <Fragment>
      {newData.map(data => (
        <div key={data.id}>
          <Title text={data.id}/>
          <VerticalChartComponent data={data.data} />
        </div>
      ))}
    </Fragment>
  )
};

VerticalChart.propTypes = {
  data: PropTypes.array
};

VerticalChart.defaultProps = {
  data: []
};

export default VerticalChart;