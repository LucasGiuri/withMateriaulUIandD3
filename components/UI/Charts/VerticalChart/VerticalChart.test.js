import renderer from 'react-test-renderer';
import VerticalChart from './VerticalChart';
import 'jest-styled-components';

describe('VerticalChart', () => {
  const data = [{
      id: 1,
      name: 'American Crew Acondicionador Uso Diario 250 ml',
      avgPriceAmazon: 200.07999999999998,
      avgPriceNotino: 212.9999999999999,
      avgPriceCarethy: 160.86,
      avgPricePerfumesclub: 284.44000000000005,
      avgPriceNutritienda: 123.89999999999995,
      avgPriceSephora: 0,
      avgPriceElcorteingles: 0,
      avgPriceFundgruben: 0,
      avgPricePrimor: 0,
      avgPriceCarrefour: 0,
      avgPriceMaquillalia: 0,
      avgPriceUlabox: 0,
      count: 30
    },
    {
      id: 2,
      name: 'American Crew Balsamo Facial SFP15 170 ml',
      avgPriceAmazon: 550.2500000000001,
      avgPriceNotino: 404.8,
      avgPriceCarethy: 584.9300000000002,
      avgPricePerfumesclub: 307.51000000000005,
      avgPriceNutritienda: 0,
      avgPriceSephora: 0,
      avgPriceElcorteingles: 0,
      avgPriceFundgruben: 0,
      avgPricePrimor: 0,
      avgPriceCarrefour: 0,
      avgPriceMaquillalia: 0,
      avgPriceUlabox: 0,
      count: 30
    }, 
    { 
      id: 3,
      name: 'American Crew ChampÃº Anticaida 250 ml',
      avgPriceAmazon: 0,
      avgPriceNotino: 0,
      avgPriceCarethy: 216.0499999999999,
      avgPricePerfumesclub: 174.99999999999997,
      avgPriceNutritienda: 272.99999999999994,
      avgPriceSephora: 0,
      avgPriceElcorteingles: 0,
      avgPriceFundgruben: 0,
      avgPricePrimor: 328.49999999999983,
      avgPriceCarrefour: 328.49999999999983,
      avgPriceMaquillalia: 0,
      avgPriceUlabox: 0,
      count: 30,
    }
  ];

  it('VerticalChart basic render snapshoot', () => {
    const component = renderer.create(<VerticalChart data={data} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
