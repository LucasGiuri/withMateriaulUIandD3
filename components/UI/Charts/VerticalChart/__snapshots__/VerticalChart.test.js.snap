// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalChart VerticalChart basic render snapshoot 1`] = `
<div
  style={
    Object {
      "height": 800,
      "width": "100%",
    }
  }
>
  <div
    className="recharts-responsive-container"
    id="undefined"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  >
    <div />
  </div>
</div>
`;
